internal/controllers/gateway/gateway_utils.go

```diff
@@ -5,23 +5,51 @@ import (
 	"reflect"
 	"strings"
 
-	"github.com/go-logr/logr"
-	"github.com/kong/kubernetes-ingress-controller/v2/internal/annotations"
-	"github.com/kong/kubernetes-ingress-controller/v2/internal/util"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 	gatewayv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/annotations"
 )
 
 // -----------------------------------------------------------------------------
-// Gateway Controller - Private Functions
+// Gateway Utilities
 // -----------------------------------------------------------------------------
 
 // maxConds is the maximum number of status conditions a Gateway can have at one time.
 const maxConds = 8
 
+// isGatewayScheduled returns boolean whether or not the gateway object was scheduled
+// previously by the gateway controller.
+func isGatewayScheduled(gateway *gatewayv1alpha2.Gateway) bool {
+	for _, cond := range gateway.Status.Conditions {
+		if cond.Type == string(gatewayv1alpha2.GatewayConditionScheduled) && cond.Reason == string(gatewayv1alpha2.GatewayReasonScheduled) {
+			return true
+		}
+	}
+	return false
```

Comment:
> Maybe I misunderstand some hidden complexity here, but shouldn't we test for .Status == "True" instead of .Reason == "Scheduled" after we've found the condition type?

Yes good catch we should be checking for `metav1.ConditionTrue` here :+1: 

This is now taken care of in eda82b7f62d2d7222954ddda8a3276f3cae37a6e

> .Conditions is a map, so after we've found the key, it doesn't make sense to loop through the remaining condition types, no?

It's technically possible for there to be duplicates, so we do want to go through the remaining conditions. That said, the API limits the total number of conditions at a single time to 8.